gdal/ogr/ogrgeometryfactory.cpp

```diff
@@ -6087,13 +6087,17 @@ OGRCurve* OGRGeometryFactory::curveFromLineString(
 /**
  * @brief Create geometry from GeoJson fragment.
  * @param pszJsonString The GeoJSON fragment for the geometry.
+ * @param nSize (new in GDAL 3.4) Optional length of the string
+ *              if it is not null-terminated
  * @return a geometry on success, or NULL on error.
  * @since GDAL 2.3
  */
-OGRGeometry* OGRGeometryFactory::createFromGeoJson( const char *pszJsonString )
+OGRGeometry* OGRGeometryFactory::createFromGeoJson(
+    const char *pszJsonString, size_t nSize )
 {
     CPLJSONDocument oDocument;
-    if( !oDocument.LoadMemory( reinterpret_cast<const GByte*>(pszJsonString)) )
+    if( !oDocument.LoadMemory( reinterpret_cast<const GByte*>(pszJsonString),
```

Comment:
this is a bit paranoiac, but in case someone on 64-bit would provide a > 2 GB buffer, add a ``(nSize == static_cast<size_t>(-1) || nSize < static_cast<size_t>(std::numeric_limits<int>::max()))`` check